org.eclipse.scout.rt.dataobject/src/main/java/org/eclipse/scout/rt/dataobject/migration/AbstractDoValueMigrationHandlerByMap.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2010, 2025 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.scout.rt.dataobject.migration;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Simple {@link AbstractDoValueMigrationHandler} migrating values from old to new using a {@link Map} storing all
+ * possible old to new mappings. The {@link #initMigrationMap()} method is called only once during object
+ * initialization.
+ */
+public abstract class AbstractDoValueMigrationHandlerByMap<T> extends AbstractDoValueMigrationHandler<T> {
```

Comment:
Maybe we could add a unit test? See DataObjectMigratorValueMigrationTest which includes multiple value migration handlers. We could add a migration handler similar to HouseTypeFixtureDoValueMigrationHandler_2, but based on the new AbstractDoValueMigrationHandlerByMap

Reply:
Added a test, what do you think about it?